core/dbt/include/global_project/macros/adapters/apply_grants.sql

```diff
@@ -0,0 +1,93 @@
+{% macro are_grants_copied_over_when_replaced() %}
```

Comment:
TODO before merge: pick shorter/clearer names for
- `are_grants_copied_over_when_replaced`
- `do_we_need_to_show_and_revoke_grants`

Reply:
@jtcohen6 `copy_grants` seems very reasonable.

if keeping the name same across the board seems like the best plan, i'd personally err towards what the goal of that input will be which would be should we revoke this thing, if we wanted to keep names as similar as possible but possibly make it easier to point out logs wise where we are in future debugging, maybe something like `should_revoke_bool` which is what the output thats assigned to the `should_revoke` variable that becomes our argument for apply_grants.

Reply:
> goal of that input will be which would be should we revoke this thing

Agreed — I'm thinking that the goal here is less _should we revoke_, and actually _should we calculate diffs between current + desired, then revoke &/or grant as need be_.

Whatever we pick, I think we should use the same name for this macro, and for the argument to `apply_grants` (currently named `should_revoke`).

Reply:
currently using the original `should_revoke` and `copy_grants` names till we get some other ideas, `calculate_diffs` is a very good option as well more than willing to change it later throughout as needed

Reply:
Happy with the consistency for now!